server/utils/nunjucksSetup.ts

```diff
@@ -89,12 +90,27 @@ export function registerNunjucks(app?: express.Express): Environment {
   njkEnv.addGlobal('coalesce', (...args: unknown[]) => args.find(item => item !== null && item !== undefined))
   njkEnv.addFilter('safeToString', (val: unknown) => (val !== null && val !== undefined ? val.toString() : val))
 
+  njkEnv.addFilter('find', (vals: Record<string, unknown>[], attributeName, valueToFind: unknown) =>
+    !vals ? [] : vals.find(val => val[attributeName] === valueToFind),
+  )
+  njkEnv.addFilter('filter', (vals: Record<string, unknown>[], attributeName, valueToFind: unknown) =>
+    !vals ? [] : vals.filter(val => val[attributeName] === valueToFind),
+  )
+  njkEnv.addFilter('attr', (vals: Record<string, unknown>[], attributeName) =>
```

Comment:
possibly dodgy name that I'm not super attached to but this is what some other projects call this kind of function